### PR TITLE
Python 3 compatibility error handling: use to_native(e) instead of str(e) or e.me…

### PR DESCRIPTION
**Pull request by veloutin**
##### SUMMARY
This simply replaces the use of `e.message` and `str(e)` on exceptions to use ansible's `to_native`

I had to patch elasticache as it was failing for me, and patched everything while at it.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
A bunch of different components, same fix and error handling almost everywhere.

##### ADDITIONAL INFORMATION

This is the original issue I had that prompted the fix.
```
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_elasticache_payload_sz37cu1p/ansible_elasticache_payload.zip/ansible/modules/cloud/amazon/elasticache.py", line 209, in create
  File "/home/vince/.local/lib/python3.6/site-packages/botocore/client.py", line 316, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/vince/.local/lib/python3.6/site-packages/botocore/client.py", line 626, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.InvalidParameterCombinationException: An error occurred (InvalidParameterCombination) when calling the CreateCacheCluster operation: Instance type cache.r5.xlarge can only be created in a VPC.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/vince/.ansible/tmp/ansible-tmp-1585912421.616607-191146578436763/AnsiballZ_elasticache.py", line 102, in <module>
    _ansiballz_main()
  File "/home/vince/.ansible/tmp/ansible-tmp-1585912421.616607-191146578436763/AnsiballZ_elasticache.py", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/vince/.ansible/tmp/ansible-tmp-1585912421.616607-191146578436763/AnsiballZ_elasticache.py", line 40, in invoke_module
    runpy.run_module(mod_name='ansible.modules.cloud.amazon.elasticache', init_globals=None, run_name='__main__', alter_sys=True)
  File "/usr/lib/python3.6/runpy.py", line 205, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_elasticache_payload_sz37cu1p/ansible_elasticache_payload.zip/ansible/modules/cloud/amazon/elasticache.py", line 540, in <module>
  File "/tmp/ansible_elasticache_payload_sz37cu1p/ansible_elasticache_payload.zip/ansible/modules/cloud/amazon/elasticache.py", line 527, in main
  File "/tmp/ansible_elasticache_payload_sz37cu1p/ansible_elasticache_payload.zip/ansible/modules/cloud/amazon/elasticache.py", line 165, in ensure_present
  File "/tmp/ansible_elasticache_payload_sz37cu1p/ansible_elasticache_payload.zip/ansible/modules/cloud/amazon/elasticache.py", line 212, in create
AttributeError: 'InvalidParameterCombinationException' object has no attribute 'message'

```

With this change it works.

### CODE REVIEW

**Comment by veloutin** [comment]
As an extra note, this PR is not much more than a search and replace. I have not thoroughly tested this, and have (so far) only used the elasticache module in it. I will be gradually using the other elements as much as possible, favoring this repository over the modules shipped with ansible 2.9.5 .

File: plugins/modules/ecs_taskdefinition.py [hunk 1]
```diff
@@ -277,7 +277,7 @@ def register_task(self, family, task_role_arn, execution_role_arn, network_mode,
         try:
             response = self.ecs.register_task_definition(**params)
         except botocore.exceptions.ClientError as e:
-            self.module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+            self.module.fail_json(msg=to_native(e), **camel_dict_to_snake_dict(e.response))
```
Comment (tremble) [inline plugins/modules/ecs_taskdefinition.py:280]: This module is using AnsibleAWSModule, which means we have a 'fail' method which is much easier to use and avoids the need to for the extra import.

```suggestion
            self.module.fail_json_aws(e, msg="Failed to register task")
```

File: plugins/modules/ec2_vpc_endpoint.py [hunk 2]
```diff
@@ -186,12 +186,14 @@
...
 
 from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
 from ansible_collections.amazon.aws.plugins.module_utils.ec2 import (get_aws_connection_info,
                                                                      boto3_conn,
                                                                      ec2_argument_spec,
                                                                      HAS_BOTO3,
                                                                      camel_dict_to_snake_dict,
                                                                      )
+from ansible_collections.amazon.aws.plugins.module_utils.aws.core import is_boto3_error_code
```
Comment (tremble) [inline plugins/modules/ec2_vpc_endpoint.py:196]:
```suggestion
from ansible_collections.amazon.aws.plugins.module_utils.core import is_boto3_error_code
```

Urgh, and between the initial review and now the imports got moved around.  It was fixed for the main branch, but we didn't check PRs :(

**Comment by veloutin** [comment]
@tremble I fixed it up. I grepped the rest just in case to find other refs to `module_utils.aws.core`.

Based on how long it took for me to respond to the feedback and requested changes, I don't think I would be in any position to complain about having to come back and fix a few imports! :sweat_smile:

**Comment by veloutin** [comment]
I have attempted to merge, but the increasing scope of the changes is making me slightly nervous, as the scope is increasing.
I suppose it has crossed from being a simple compat fix to a broader cleanup. 

A lot of the `to_native` calls I actually replaced by using `fail_json_aws`, except in a few spots where I wasn't sure (mostly different exceptions being caught). I also had to add some `is_boto3_error_code`.

**Pull request merged**